### ISSUE TITLE
DISPLAY_WHEN_INVISIBLE doesn't work, logic error

### ISSUE THREAD

**Issue by SergioK29** [Rosewood-Development/PlayerParticles#125]
```java
public static boolean isPlayerVisible(Player player) {
        return Setting.DISPLAY_WHEN_INVISIBLE.getBoolean() || ((NMSUtil.getVersionNumber() <= 8 || player.getGameMode() != GameMode.SPECTATOR)
                && !player.hasPotionEffect(PotionEffectType.INVISIBILITY)) || player.getMetadata("vanished").stream().noneMatch(MetadataValue::asBoolean);
    }
```

if player doesn't have vanished metadata, they will always show particles even if in invis due to logic error.
    
TIP: inlining code is satisfying but when you have this many conditions using early returns is easier to read
    
too lazy to open pr, here is correct code:
    
```java
     public static boolean isPlayerVisible(Player player) {
        if (Setting.DISPLAY_WHEN_INVISIBLE.getBoolean()) {
            return true;
        }
        if (NMSUtil.getVersionNumber() >= 8 && player.getGameMode() == GameMode.SPECTATOR) {
            return false;
        }
        if (player.hasPotionEffect(PotionEffectType.INVISIBILITY)) {
            return false;
        }
        if (player.getMetadata("vanished").stream().anyMatch(MetadataValue::asBoolean)) {
            return false;
        }

        return true;
    }
```

**Comment by Esophose** [comment]
I swear I just changed this due to another issue that was open.
Other commit "fix": 95d74fd2cc6b296f2d652c0ee29d29644a44ee8b
Other issue: #117
Perhaps I should actually test it this time. I'll fix this shortly.

**Comment by Esophose** [comment]
Should be fixed in 7e57e49da53710d0198e83ab450f8dafd7e8acf2
